src/http.js

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable new-cap */
+'use strict';
+
+const _ = require('lodash');
+const jayson = require('jayson');
+
+const RippleAPI = require('./api').RippleAPI;
+
+function createHTTPServer(options, httpPort) {
+  const rippleAPI = new RippleAPI(options);
+
+  const methodNames = _.filter(_.keys(RippleAPI.prototype), k => {
+    return typeof RippleAPI.prototype[k] === 'function'
+    && k !== 'connect'
+    && k !== 'disconnect'
+    && k !== 'constructor'
+    && k !== 'RippleAPI';
+  });
+
+  function applyPromiseWithCallback(fnName, callback, args) {
+    try {
+      Promise.resolve(rippleAPI[fnName].apply(rippleAPI, args))
+        .then(res => callback(null, res))
+        .catch(err => {
+          callback({code: 99, message: err.message, data: {name: err.name}});
+        });
+    } catch (err) {
+      callback({code: 99, message: err.message, data: {name: err.name}});
+    }
+  }
+
+  const methods = {};
+  _.forEach(methodNames, fn => {
+    methods[fn] = jayson.Method((args, cb) => {

```

Comment:
Ah, probably because this is using a single function argument. I think we would need to dynamically de-promisify while keeping the same method arguments.